trl/trainer/callbacks.py

```diff
@@ -281,15 +286,27 @@ def on_train_begin(self, args: TrainingArguments, state: TrainerState, control:
             )
             # Compute initial win rate as a reference point
             completions = list(zip(self.ref_completions, self.ref_completions))
-            winner_indices = self.judge.judge(prompts, completions, self.shuffle_order)
-            prompts = gather_object(prompts)
-            completions = gather_object(completions)
-            winner_indices = gather_object(winner_indices)
+            if self.use_soft_judge:
+                ref_win_probs = self.judge.judge(prompts, completions, self.shuffle_order, return_scores=True)
+                winner_indices = [0 if score > 0.5 else 1 for score in ref_win_probs]
+                prompts = gather_object(prompts)
+                completions = gather_object(completions)
+                ref_win_probs = gather_object(ref_win_probs)
+            else:
+                winner_indices = self.judge.judge(prompts, completions, self.shuffle_order)
+                prompts = gather_object(prompts)
+                completions = gather_object(completions)
+                winner_indices = gather_object(winner_indices)
 
         # Logging
         if self.trainer.accelerator.is_main_process:
-            win_rate = sum(winner_idx == 1 for winner_idx in winner_indices) / len(winner_indices)
-            self.trainer.log({"eval_win_rate": win_rate})
+            if self.use_soft_judge:
+                soft_win_rate = 1.0 - sum(ref_win_probs) / len(ref_win_probs)
```

Comment:
It's not a “rate” as such, but rather an average probability of winning, or something like that. Do you have any better naming?

Reply:
`avg_win_prob` ?